src/DsaLocal.cc

```diff
@@ -1084,6 +1087,46 @@ void LocalAnalysis::runOnFunction(Function &F, Graph &g) {
   );
   // clang-format on
 
+  /// Mark incompelete nodes after the pass
+  /// based on the code in remove_dead
+  /// TODO: doing it inside remove_dead maybe more efficient
+  std::set<const Node *> reachable;
+  // --- collect all nodes referenced by formal parameters
+  for (auto &kv : llvm::make_range(g.formal_begin(), g.formal_end())) {
```

Comment:
This does not go deep enough. It only marks nodes that are directly passed as formal arguments.
But nodes that are reachable from them should also be marked. 
For example, I pass `p` to a procedure, and procedure uses `p->next->next->next->data`